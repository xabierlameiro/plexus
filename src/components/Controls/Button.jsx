import React from 'react'
import PropTypes from 'prop-types'
import './button.scss'

const Button = ({ classes, disabled = true, text, handler, icon }) => {
    return (
        <button
            data-testid="button"
            className={classes}
            onClick={handler}
            disabled={!disabled}
        >
            {text}
            {icon}
        </button>
    )
}

Button.propTypes = {
    classes: PropTypes.string,
    disabled: PropTypes.bool,
    text: PropTypes.string,
    handler: PropTypes.func,
    icon: PropTypes.node,
}

export default Button
