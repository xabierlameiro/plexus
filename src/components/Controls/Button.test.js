import Button from './Button'
import '@testing-library/jest-dom/extend-expect'

import { render, cleanup } from '@testing-library/react'

import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
    const div = document.createElement('div')
    render(<Button />, div)
})

it('renders button correctly', () => {
    const { getByTestId } = render(<Button text="Click me" />)
    expect(getByTestId('button')).toHaveTextContent('Click me')
})

it('matches snapshot', () => {
    const tree = renderer
        .create(<Button classes="terciary" text="Click me" />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})
