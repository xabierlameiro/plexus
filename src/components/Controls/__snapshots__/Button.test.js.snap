// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="terciary"
  data-testid="button"
  disabled={false}
>
  Click me
</button>
`;
